File: comment-service/src/main/kotlin/com/instagram_clone/comment_service/util/ConfigHelper.java
package com.instagram_clone.comment_service.util;

import io.vertx.config.ConfigRetriever;
import io.vertx.core.Future;
import io.vertx.core.Promise;
import io.vertx.core.json.JsonObject;

public class ConfigHelper {

  public static Future<JsonObject> getConfig(ConfigRetriever retriever) {
    Promise<JsonObject> promise = Promise.promise();
    retriever.getConfig((ar) -> {
      if (ar.succeeded()) {
        promise.complete(ar.result());
      } else {
        promise.fail(ar.cause());
      }
    });
    return promise.future();
  }
}

File: comment-service/src/main/kotlin/com/instagram_clone/comment_service/util/ConfigHelper.kt
package com.instagram_clone.comment_service.util

import io.vertx.core.json.JsonObject
import java.lang.ClassCastException

/**
 * This function is used in cases where the config.json has values as strings
 * and for some reason the same values read from env variables in docker container
 * turn to doubles.
 */
fun retrieveProblematicString(config: JsonObject, key: String) : String {
  return try {
    config.getString(key)
  } catch (e: ClassCastException) {
    config.getDouble(key).toString().substringBeforeLast(".")
  }
}
